dashboard/app/routes/optimization/fine-tuning/route.tsx

```diff
@@ -0,0 +1,334 @@
+    // TODO (Viraj, important!): fix the status here.
+    const status = updatedJob.status();
+    const modelProvider = updatedJob.provider();
+
+    return {
+      status,
+      result,
+      modelProvider,
+    };
+  } catch (error) {
+    return {
+      status: "error",
+      error: error instanceof Error ? error.message : "Unknown error",
+    };
+  }
+}
+
+// The action actually launches the fine-tuning job.
+export async function action({ request }: Route.ActionArgs) {
+  const formData = await request.formData();
+  const serializedFormData = formData.get("data");
+  if (!serializedFormData || typeof serializedFormData !== "string") {
+    throw new Error("Form data must be provided");
+  }
+
+  const jsonData = JSON.parse(serializedFormData);
+  const validatedData = SFTFormValuesSchema.parse(jsonData);
+
+  const job = await launch_sft_job(validatedData);
+  jobStore[validatedData.jobId] = job;
+
+  return redirect(`/optimization/fine-tuning/${validatedData.jobId}`);
+}
+
+// Renders the fine-tuning form and status info.
+export default function FineTuning({ loaderData }: Route.ComponentProps) {
+  const config = useConfig();
+  const { status, result, modelProvider } = loaderData;
+  const revalidator = useRevalidator();
+
+  // If running, periodically poll for updates on the job
+  useEffect(() => {
+    if (status === "running") {
+      setSubmissionPhase("pending");
+      setIsSubmitted(true);
+      const interval = setInterval(() => {
+        revalidator.revalidate();
+      }, 10000);
+      return () => clearInterval(interval);
+    }
+  }, [status, revalidator]);
+
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [submissionPhase, setSubmissionPhase] = useState<
+    "idle" | "submitting" | "pending" | "complete"
+  >("idle");
+  const finalResult = result
+    ? dump_model_config(get_fine_tuned_model_config(result, modelProvider))
+    : null;
+  if (finalResult && submissionPhase !== "complete") {
+    setSubmissionPhase("complete");
+  }
+
+  return (
+    <div className="min-h-screen bg-gray-50">
+      <main className="p-4">
+        <h2 className="scroll-m-20 border-b pb-2 text-3xl font-semibold tracking-tight first:mt-0">
+          Fine-Tuning
+        </h2>
+        {status === "idle" && (
+          <FineTuningForm
+            config={config}
+            submissionPhase={submissionPhase}
+            setSubmissionPhase={setSubmissionPhase}
+            isSubmitted={isSubmitted}
+          />
+        )}
+
+        {loaderData && (
+          <div className="p-4 bg-gray-100 rounded-lg mt-4">
+            <div className="mb-2 font-medium">
+              Loader Data (Last Updated: {new Date().toLocaleTimeString()})
+            </div>
+            <Textarea
+              value={JSON.stringify(loaderData, null, 2)}
+              className="w-full h-48 resize-none bg-transparent border-none focus:ring-0"
+              readOnly
+            />
+          </div>
+        )}
```

Comment:
Only display this after submission, now while filling the form.

Reply:
sgtm